src/Engine/ProtoCore/Lang/FunctionGroup.cs

```diff
@@ -307,6 +335,86 @@ public void CopyPublic(List<FunctionEndPoint> rhs)
             return ret;
         }
 
+        internal static bool CheckInvalidArrayCoersion(CLRFunctionEndPoint fep, List<CLRStackValue> reducedSVs, MSILRuntimeCore runtimeCore, bool allowArrayPromotion)
+        {
+            ClassTable classTable = runtimeCore.ClassTable;
+
+            for (int i = 0; i < reducedSVs.Count; i++)
+            {
+                Type typ = fep.FormalParams[i].ProtoInfo;
+                if (typ.UID == (int)ProtoCore.PrimitiveType.InvalidType)
+                    return true;
+
+                if (!typ.IsIndexable)
+                    continue; //It wasn't an array param, skip
+
+                //Compute the type of target param
+                if (!allowArrayPromotion)
+                    Validity.Assert(reducedSVs[i].IsEnumerable, "This should be an array otherwise this shouldn't have passed previous tests");
+
+                if (!allowArrayPromotion)
+                {
+                    if (typ.rank != ArrayUtils.GetMaxRankForArray(reducedSVs[i]) &&
```

Comment:
I think braces would make these nested if statements less prone to misreading or typos.